dnf/cli/output.py

```diff
@@ -1628,7 +1628,13 @@ def historyListCmd(self, tids):
         fmt = "%s | %s | %s | %s | %s"
         if len(uids) == 1:
             name = _("Command line")
-            name_width = self.term.columns - 55 if self.term.columns > 79 else 24
+            if self.term.real_columns is None:
+                name_width = (
+                        24 if not transactions
+                        else max([len(t.cmdline or '') for t in transactions])
+                        )
+            else:
+                name_width = self.term.columns - 55 if self.term.columns > 79 else 24
```

Comment:
What about to decrease the number of calls self.term.real_columns or self.term.columns?
```
            real_width = self.term.real_columns
            if real_width is None:
                name_width = (
                        24 if not transactions
                        else max([len(t.cmdline or '') for t in transactions])
                        )
            else:
                name_width = real_width - 55 if real_width > 79 else 24
```